client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ClientListenerNotifier.java

```diff
@@ -67,7 +68,16 @@ protected ClientListenerNotifier(ExecutorService executor, Codec codec, Marshall
    }
 
    public static ClientListenerNotifier create(Codec codec, Marshaller marshaller) {
-      return new ClientListenerNotifier(Executors.newCachedThreadPool(), codec, marshaller);
+      ExecutorService executor = Executors.newCachedThreadPool(getRestoreThreadNameThreadFactory());
+      return new ClientListenerNotifier(executor, codec, marshaller);
+   }
+
+   private static ThreadFactory getRestoreThreadNameThreadFactory() {
+      return r -> new Thread(() -> {
+         final String originalName = Thread.currentThread().getName();
+         r.run();
+         Thread.currentThread().setName(originalName);
```

Comment:
Overriding the default thread factory looses some of the initial behaviour. The original TF ensures the thread priority is normal instead of being inherited from the caller. Do we care?